firestore.rules

```diff
@@ -0,0 +1,82 @@
+rules_version = '2';
+service cloud.firestore {
+  match /databases/{database}/documents {
+
+  	function isAdmin(env) {
+      // Check Admin Privileges.
+      return get(/databases/$(database)/documents/env/$(env)/admins/$(request.auth.uid)).data.secret == "!_xEqyCw3Y.6*wrW32EkCz6GUUn7";
+    }
+
+    match /env/{env}/{document=**} {
+    	// Default document rules.
+    	allow read: if request.auth.uid != null;
+      allow write: if false;
+    }
+
+    match /env/{env} { // PRODUCTION or DEVELOPMENT
+
+    	match /admins/{admin} {
+      	allow read, write: if false;
+      }
+
+    	match /children/{child} {
+      	allow write: if request.auth.uid != null;
+      }
+
+      match /classes/{class} {
+      	allow write: if get(resource.data.teacher).id == request.auth.uid || isAdmin(env);
+        allow create, update: if request.auth.uid != null;
```

Comment:
Now that class assignment is handled by the cloud functions, we no longer need the `update` permissions available to all users, and it should be restricted to only the teacher who created the class.

## Suggestion
```
match /classes/{class} {
    allow update: if get(resource.data.teacher).id == request.auth.uid;
    allow create: if request.auth.uid != null;
    allow read: true;
    allow write: false;
}
```

Reply:
Resolved in commit [1c88041](https://github.com/codecontest-org/app/pull/3/commits/1c880416cda627e75968db8cb9f5eb78e622eb55).